dashboard/app/dsl/user_id_replace_dsl.rb

```diff
@@ -0,0 +1,2 @@
+class UserIdReplaceDSL < ExternalDSL
+end
```

Comment:
The goal of this was to be just like External, with this added functionality. We probably could just add this functionality to External itself and be fine. Not using External would essentially meaning porting External to a non DSLDefine type, which might be goodness, but is a lot more work and out of scope.